compiler/modules/bir/bir.bal

```diff
@@ -62,8 +73,16 @@ public type InternalSymbol readonly & record {|
 
 public type Symbol InternalSymbol|ExternalSymbol;
 
-public type FunctionRef readonly & record {|
-    Symbol symbol;
+public type FunctionRef ExternalFunctionRef | InternalFunctionRef;
```

Comment:
@jclark One problem with completely factoring `FunctionRef` to be the same between Anonymous function and function definitions is we can't give an index to langlib functions [here](https://github.com/ballerina-platform/nballerina/blob/84ccc96b98b72e86b4cfef68400cdcb1439cb49e/compiler/modules/front/expr.bal#L1631), since we don't really have an actual module for langlib modules. Also, I am not sure if is a good idea to use an index for external functions. As such I decided to factor FunctionRef into `ExternalFunctionRef` (which will use a symbol) and `InternalFunctionRef` (common for both anonymous functions and normal ones)